bundles/binding/org.openhab.binding.plugwise/src/main/java/org/openhab/binding/plugwise/internal/PlugwiseBinding.java

```diff
@@ -68,142 +74,170 @@ protected void removeBindingProvider(PlugwiseBindingProvider bindingProvider) {
-                    if (stick == null) {
+        if (port == null) {
+            return null;
+        }
 
-                        String configKey = matcher.group(2);
-                        String value = (String) config.get(key);
+        Stick stick = new Stick(port, this);
+        logger.info("Plugwise added Stick connected to serial port {}", port);
 
-                        if ("port".equals(configKey)) {
-                            stick = new Stick(value, this);
-                            logger.info("Plugwise added Stick connected to serial port {}", value);
-                        } else if ("interval".equals(configKey)) {
-                            // do nothing for now. we will set in the second run
-                        } else if ("retries".equals(configKey)) {
-                            // do nothing for now. we will set in the second run
-                        }
+        String interval = (String) config.get("stick.interval");
+        if (interval != null) {
+            stick.setInterval(Integer.valueOf(interval));
+            logger.info("Setting the interval to send ZigBee PDUs to {} ms", interval);
+        }
 
-                        else {
-                            throw new ConfigurationException(configKey,
-                                    "the given configKey '" + configKey + "' is unknown");
-                        }
-                    }
-                }
+        String retries = (String) config.get("stick.retries");
+        if (retries != null) {
+            stick.setRetries(Integer.valueOf(retries));
+            logger.info("Setting the maximum number of attempts to send a message to ", retries);
+        }
+
+        return stick;
+    }
+
+    private void setupNonStickDevices(Dictionary<String, ?> config) {
+
+        Set<String> deviceNames = getDeviceNamesFromConfig(config);
 
+        for (String deviceName : deviceNames) {
+            if ("stick".equals(deviceName)) {
+                continue;
             }
 
-            if (stick != null) {
-                // re-run through the configuration and setup the remaining devices
-                keys = config.keys();
-                while (keys.hasMoreElements()) {
+            if (stick.getDeviceByName(deviceName) != null) {
+                continue;
+            }
 
-                    String key = (String) keys.nextElement();
+            String MAC = (String) config.get(deviceName + ".mac");
+            if (MAC == null || MAC.equals("")) {
+                logger.error("Plugwise can not add device with name {} without a MAC address", deviceName);
+            } else if (stick.getDeviceByMAC(MAC) != null) {
+                logger.error(
+                        "Plugwise can not add device with name: {} and MAC address: {}, "
+                                + "the same MAC address is already used by device with name: {}",
+                        deviceName, MAC, stick.getDeviceByMAC(MAC).name);
```

Comment:
The two `logger.error` above both seem to be more appropriately `logger.warn` unless system stability is affected.